File: src/co/nums/intellij/aem/htl/file/HtlFileTypeFactory.java
package co.nums.intellij.aem.htl.file;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class HtlFileTypeFactory extends FileTypeFactory {

	@Override
	public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
		fileTypeConsumer.consume(HtlFileType.INSTANCE, HtlFileType.DEFAULT_EXTENSION);
	}

}

File: src/co/nums/intellij/aem/htl/file/HtlFileTypeFactory.kt
package co.nums.intellij.aem.htl.file

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class HtlFileTypeFactory : FileTypeFactory() {

    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) =
            fileTypeConsumer.consume(HtlFileType, HtlFileType.defaultExtension)

}
